graviton/blackbox.py

```diff
@@ -174,406 +196,386 @@ def _partial_encode_assert(
-            txn_params=cls.transaction_params(
-                sender=ZERO_ADDRESS if sender is None else sender,
-                sp=cls.SUGGESTED_PARAMS if sp is None else sp,
-                note=note,
-                lease=lease,
-                rekey_to=rekey_to,
-                receiver=ZERO_ADDRESS if receiver is None else receiver,
-                amt=0 if amt is None else amt,
-                close_remainder_to=close_remainder_to,
-            ),
-        )
 
-    @classmethod
-    def dryrun_app_pair_on_sequence(
-        cls,
-        algod: AlgodClient,
-        teal_and_method1: TealAndMethodType,
-        teal_and_method2: TealAndMethodType,
-        inputs: List[Sequence[PyTypes]],
-        *,
-        is_app_create: bool = False,
-        on_complete: OnComplete = OnComplete.NoOpOC,
-        dryrun_accounts: List[DryRunAccountType] = [],
-        omit_method_selector: Optional[bool] = False,
-        validation: bool = True,
-    ) -> Tuple[Sequence["DryRunInspector"], Sequence["DryRunInspector"]]:
-        return tuple(  # type: ignore
-            cls.dryrun_multiapps_on_sequence(
-                algod=algod,
-                multi_teal_method_pairs=[teal_and_method1, teal_and_method2],
-                inputs=inputs,
-                is_app_create=is_app_create,
-                on_complete=on_complete,
-                dryrun_accounts=dryrun_accounts,
-                omit_method_selector=omit_method_selector,
-                validation=validation,
-            )
-        )
+class DryRunExecutor:
 
-    @classmethod
-    def dryrun_multiapps_on_sequence(
-        cls,
-        algod: AlgodClient,
-        multi_teal_method_pairs: List[TealAndMethodType],
-        inputs: List[Sequence[PyTypes]],
-        *,
-        is_app_create: bool = False,
-        on_complete: OnComplete = OnComplete.NoOpOC,
-        dryrun_accounts: List[DryRunAccountType] = [],
-        omit_method_selector: Optional[bool] = False,
-        validation: bool = True,
-    ) -> List[Sequence["DryRunInspector"]]:
-        def runner(teal_method_pair):
-            teal = teal_method_pair[0]
-            abi_method = None
-            if len(teal_method_pair) > 1:
-                abi_method = teal_method_pair[1]
-
-            return cls.dryrun_app_on_sequence(
-                algod=algod,
-                teal=teal,
-                inputs=inputs,
-                abi_method_signature=abi_method,
-                omit_method_selector=omit_method_selector,
-                validation=validation,
-                is_app_create=is_app_create,
-                on_complete=on_complete,
-                dryrun_accounts=dryrun_accounts,
-            )
+    """Methods to package up and kick off dry run executions"""
 
-        return list(map(runner, multi_teal_method_pairs))
+    # for usage convenience, copy constants over into the class
+    CREATION_APP_CALL = CREATION_APP_CALL
+    EXISTING_APP_CALL = EXISTING_APP_CALL
```

Comment:
wondering what these constants get copied into the `DryRunExecutor`, is it like this line https://github.com/algorand/pyteal/pull/628/files#diff-6e53c9f2d18240e39cc4cf7fbab2086bc6d843229ed1e1c3a692238284e9c190R33 in https://github.com/algorand/pyteal/pull/628?

Reply:
Yes, exactly!